pantheon/src/test/java/tech/pegasys/pantheon/cli/PantheonCommandTest.java

```diff
@@ -2612,38 +2608,31 @@ public void errorIsRaisedIfStaticNodesAreNotWhitelisted() throws IOException {
   public void pendingTransactionRetentionPeriod() {
     final int pendingTxRetentionHours = 999;
     parseCommand("--tx-pool-retention-hours", String.valueOf(pendingTxRetentionHours));
-
-    verify(mockControllerBuilder)
-        .transactionPoolConfiguration(transactionPoolConfigurationArgumentCaptor.capture());
-    assertThat(transactionPoolConfigurationArgumentCaptor.getValue().getPendingTxRetentionPeriod())
-        .isEqualTo(pendingTxRetentionHours);
-
+    verify(mockTransactionPoolConfigurationBuilder)
+        .pendingTxRetentionPeriod(pendingTxRetentionHours);
     assertThat(commandOutput.toString()).isEmpty();
     assertThat(commandErrorOutput.toString()).isEmpty();
   }
 
   @Test
   public void txMessageKeepAliveSeconds() {
     final int txMessageKeepAliveSeconds = 999;
-    parseCommand("--tx-pool-keep-alive-seconds", String.valueOf(txMessageKeepAliveSeconds));
-
-    verify(mockControllerBuilder)
-        .transactionPoolConfiguration(transactionPoolConfigurationArgumentCaptor.capture());
-    assertThat(transactionPoolConfigurationArgumentCaptor.getValue().getTxMessageKeepAliveSeconds())
-        .isEqualTo(txMessageKeepAliveSeconds);
-
+    parseCommand(
+        "--Xincoming-tx-messages-keep-alive-seconds", String.valueOf(txMessageKeepAliveSeconds));
+    verify(mockTransactionPoolConfigurationBuilder).build();
```

Comment:
You're not actually checking that the value 999 is set.

Reply:
Yes. We have not yet the mechanism to do this kind of assertion with unstable options. I suggest we think about it in another PR because it is out of scope of this PR.

Reply:
We found a way to test it using a real object instead of the mock